src/plugins/auto/plugin.cpp

```diff
@@ -496,39 +496,103 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     std::mutex load_mutex;
     std::vector<Task> loads;
     std::once_flag readNetworkFlag;
-    for (auto& p : metaDevices) {
-        loads.push_back([&]() {
-            auto tmpiter = fullConfig.find(CONFIG_KEY(ALLOW_AUTO_BATCHING));
-            if (tmpiter != fullConfig.end()) {
-                if (tmpiter->second == PluginConfigParams::NO)
-                    multiSContext->_batchingDisabled = true;
-                p.config.insert({tmpiter->first, tmpiter->second});
+    std::vector<DeviceInformation> cpuDevice;
```

Comment:
do we have situation when more than one CPU device? maybe if not, we do not need vector here, as below many logic only use the first element in this vector.

Reply:
Modified to not use vector for CPU